hack/generate-manifest.sh

```diff
@@ -138,7 +144,21 @@ if $KIND; then
     $KUSTOMIZE edit add patch installCni.yml
 fi
 
-$KUSTOMIZE build
+if $IPSEC; then
+    cp ../../patches/ipsec/*.yml .
+
+    # create the K8s Secret for IPSec PSK
+    $KUSTOMIZE edit add resource ipsecSecret.yml
+    # add the PSK env variable to the antrea-agent container
+    $KUSTOMIZE edit add patch pskEnv.yml
+    # add a container to the Agent DaemonSet to run IPSec daemons
+    $KUSTOMIZE edit add patch ipsecContainer.yml
+
+    # set config option enableIPSecTunnel to true
+    $KUSTOMIZE build | sed 's/#enableIPSecTunnel: false/enableIPSecTunnel: true/'
```

Comment:
you shouldn't use `sed` here. It "breaks" kustomize ConfigMap support because now the generated hash no longer matches the contents of the ConfigMap. Ideally we would have a proper ConfigMap with a list of parameters and not just a JSON file, but for now we can generate the JSON in `generate-manifest.sh` I think.

Reply:
What you think if I use sed or whatever to generate a kustomize path with generate-manifest.sh that takes the original config file with the revisions for IPSec?

Reply:
Yes, that's what I had in mind: generate the conf file from a template upfront based on the options (ipsec / kind) and use it with kustomize. I can draft this out in my PR which refactors the overlays.